### DISCUSSION TITLE
Config format discussion

### DISCUSSION THREAD

**Discussion by electron271** [MagmaWM/MagmaWM#52] [Q&A] [answered]
TOML is not a good pick due to being hard for the user, and hard for us to parse, so we are moving to .conf, this discussion is for discussing this change.

**Comment by eepy-goddess** [comment]
Toml isn't hard at all to parse, and it's much easier for the user than RON

**Reply by ghost** [reply]
it is hard to parse in a wm context

**Reply by electron271** [reply]
it is in terms of window manager, toml is overall not a good choice for this.

**Reply by eepy-goddess** [reply]
I already made a good toml config generator and parser for my own compositor, but ok

**Reply by electron271** [reply]
doesnt mean its good for this wm

**Reply by vladimir-lu** [reply]
What are the reasons it's hard to parse? Is it because you don't want a dependency on many crates?

**Reply by electron271** [reply]
my personal worry is keybindings, and i find toml a bit annoying to parse, could be just me though

**Reply by electron271** [reply]
i also feel that conf would be a better solution for this

**Reply by ghost** [reply, marked as answer]
toml would be easier for us but would be realy bad for the end user
Conf would be harder since we gotta design our own config file and parse it but that also has a alot of benefits

**Reply by electron271** [reply]
yeah toml may be easier in some ways but honestly it sucks for the end user

**Comment by electron271** [comment]
also why is the icon for q and a a praying emoji

**Reply by ghost** [reply]
idfk ask microsoft

**Comment by luckyez** [comment]
I was wondering if it's possible to use multiple config formats, if it's possible it would be a nice thing to have for the users.

**Reply by electron271** [reply]
this *is* possible, but would be hard to implement, and i also feel it would make things harder for users

**Comment by eepy-goddess** [comment]
our users would be linux users, bsd users, and people smart enough to get a wayland compositor working on windows, so user friendliness won't be very significant to the development of this compositor....

**Reply by electron271** [reply]
still important

**Reply by eepy-goddess** [reply]
I know toml isn't the most user friendly, but still way better than json

**Reply by electron271** [reply]
nobody uses json for window manager configuration

or really configuration at all lmao

**Reply by eepy-goddess** [reply]
I mean, waybar uses json for its config :kekw:

**Reply by electron271** [reply]
thats a bar, and thats like one thing

**Reply by eepy-goddess** [reply]
Still configuration :trollface:

**Reply by electron271** [reply]
the point is we are not using toml

**Reply by eepy-goddess** [reply]
Only hackedOS and sakura can decide where the project will be going, and they agreed to use toml i believe

**Reply by electron271** [reply]
1. not exactly all devs have control lol
2. and no they and me were the people who decided to NOT use toml

**Reply by eepy-goddess** [reply]
Ok, i give up

**Reply by electron271** [reply]
why are you insisting on toml so much anyway

**Reply by eepy-goddess** [reply]
I just like toml

**Comment by Henry-Hiles** [comment]
What about json?

**Reply by HackedOS** [reply]
No, just no, pls no. i beg u pls don't. dont even think about it

**Reply by electron271** [reply]
hell nah
